ksqldb-engine/src/test/java/io/confluent/ksql/physical/pull/operators/ProjectOperatorTest.java

```diff
@@ -0,0 +1,356 @@
+  @Mock
+  private SelectValueMapper<Object> selectValueMapper;
+
+  @Test
+  public void shouldProjectAllColumnsWhenSelectStarNonWindowed() {
+    // Given:
+    final ProjectOperator projectOperator = new ProjectOperator(
+        ksqlConfig,
+        metaStore,
+        logger,
+        mat,
+        logicalNode,
+        outputSchema,
+        true,
+        false,
+        false,
+        selectValueMapperFactorySupplier);
+    projectOperator.addChild(child);
+    final Row row = Row.of(
+        SCHEMA,
+        A_KEY,
+        GenericRow.genericRow("a", "b"),
+        A_ROWTIME
+    );
+    when(child.next()).thenReturn(row);
+    projectOperator.open();
+
+    // Then:
+    final List<Object> rowList = new ArrayList<>();
+    row.key().schema().fields().stream().map(row.key()::get).forEach(rowList::add);
+    rowList.addAll(row.value().values());
+    assertThat(projectOperator.next(), is(rowList));
+  }
+
+  @Test
+  public void shouldProjectAllColumnsWhenSelectStarWindowed() {
+    // Given:
+    final ProjectOperator projectOperator = new ProjectOperator(
+        ksqlConfig,
+        metaStore,
+        logger,
+        mat,
+        logicalNode,
+        outputSchema,
+        true,
+        true,
+        false,
+        selectValueMapperFactorySupplier);
+    projectOperator.addChild(child);
+    final WindowedRow windowedRow = WindowedRow.of(
+        SCHEMA,
+        new Windowed<>(A_KEY, STREAM_WINDOW),
+        GenericRow.genericRow("a", "b"),
+        A_ROWTIME
+    );
+    when(child.next()).thenReturn(windowedRow);
+    projectOperator.open();
+
+    // Then:
+    final List<Object> rowList = new ArrayList<>();
+    windowedRow.key().schema().fields().stream().map(windowedRow.key()::get).forEach(rowList::add);
+    rowList.add(windowedRow.window().get().start().toEpochMilli());
+    rowList.add(windowedRow.window().get().end().toEpochMilli());
+    rowList.addAll(windowedRow.value().values());
+    assertThat(projectOperator.next(), is(rowList));
+  }
+
+  @Test
+  public void shouldCallTransformWithCorrectArguments() {
+    // Given:
+    final ProjectOperator projectOperator = new ProjectOperator(
+        ksqlConfig,
+        metaStore,
+        logger,
+        mat,
+        logicalNode,
+        SCHEMA,
+        false,
+        false,
+        true,
+        selectValueMapperFactorySupplier);
+    projectOperator.addChild(child);
+    final Row row = Row.of(
+        SCHEMA,
+        A_KEY,
+        GenericRow.genericRow("a", "b"),
+        A_ROWTIME
+    );
+    when(child.next()).thenReturn(row);
+    when(selectValueMapperFactorySupplier.create(any(), any(), any(), any()))
+        .thenReturn(selectValueMapper);
+    when(selectValueMapper.getTransformer(logger)).thenReturn(transformer);
+    when(transformer.transform(any(), any(), any())).thenReturn(GenericRow.genericRow("k", "a", "b"));
+    when(mat.schema()).thenReturn(SCHEMA);
+    projectOperator.open();
+    projectOperator.next();
```

Comment:
nit: this applies to all tests, but this is what belongs in the `//When:` part (it makes it easy to identify what is setup and what the test is actually doing. I know this PR has gone on for a while so I won't ask you to refactor all the tests, but going forward it makes tests really easy to read when then `//When` portion contains the code that you're testing.

For example:
```java
// When:
List<?> result = projectOperator.next();

// Then:
final List<Object> expected = new ArrayList<>();
windowedRow.key().schema().fields().stream().map(windowedRow.key()::get).forEach(expected::add);
expected.add(windowedRow.window().get().start().toEpochMilli());
expected.add(windowedRow.window().get().end().toEpochMilli());
expected.addAll(windowedRow.value().values());
assertThat(result, is(expected)); 
```

Compare that with:
```java
// Then:
final List<Object> expected = new ArrayList<>();
windowedRow.key().schema().fields().stream().map(windowedRow.key()::get).forEach(expected::add);
expected.add(windowedRow.window().get().start().toEpochMilli());
expected.add(windowedRow.window().get().end().toEpochMilli());
expected.addAll(windowedRow.value().values());
assertThat(result, is(rowList)); 
assertThat(projectOperator.next(), is(expected)); 
```

For the second one I need to dig in to figure out what is the production code being called.